PyPoE/cli/exporter/wiki/parsers/item.py

```diff
@@ -4153,42 +4166,57 @@ def export_map(self, parsed_args):
                 wiki_message='Map exporter',
             )
 
+            # Export map icons
             if parsed_args.store_images:
-                if atlas_node is None or \
-                        not atlas_node['ItemVisualIdentityKey']['DDSFile']:
+                # Warn about and skip maps that aren't on the atlas and may not exist.
+                if atlas_node is None and base_item_type['Id'] not in MAPS_IN_SERIES_BUT_NOT_ON_ATLAS:
+                    warnings.warn(
+                        f"{base_item_type['Name']} ({base_item_type['Id']}) is not currently on the Atlas"
+                    )
+                    continue
+
+                # Warn about and skip maps that are on atlas but have no icon.
+                elif atlas_node is not None and not atlas_node['ItemVisualIdentityKey']['DDSFile']:
                     warnings.warn(
                         'Missing 2d art inventory icon for item "%s"' %
                         base_item_type['Name']
                     )
                     continue
-
+                
                 ico = os.path.join(self._img_path, name + ' inventory icon.dds')
 
+                # If the atlas doesn't point to an icon, use the base_item_type for the icon.
+                if atlas_node is not None:
+                    dds_file_path = atlas_node['ItemVisualIdentityKey']['DDSFile']
+                else:
+                    dds_file_path = base_item_type['ItemVisualIdentityKey']['DDSFile']
+
+                # Save off the map's icon (which still needs to be layered onto the base map)
                 self._write_dds(
-                    data=self.file_system.get_file(
-                        atlas_node['ItemVisualIdentityKey']['DDSFile']),
+                    data=self.file_system.get_file(dds_file_path),
                     out_path=ico,
                     parsed_args=parsed_args,
                 )
-
                 ico = ico.replace('.dds', '.png')
 
-                color = None
-                if 5 < tier <= 10:
-                    color = self._MAP_COLORS['mid tier']
-                elif 10 < tier <= 15:
-                    color = self._MAP_COLORS['high tier']
-                if color:
+                # Recolor the map icon if appropriate and layer the map icon with the base icon.
+                if base_item_type['Id'] not in MAPS_TO_SKIP_COLORING_AND_COMPOSITING:
+                    color = None
+                    if 5 < tier <= 10:
+                        color = self._MAP_COLORS['mid tier']
+                    elif 10 < tier <= 15:
+                        color = self._MAP_COLORS['high tier']
+                    if color:
+                        os.system(
+                            '''magick convert "%s" -fill rgb(%s) -colorize 100 "%s"''' % (
+                            ico, color, ico
+                        ))
```

Comment:
Not entirely sure whether this is just a mistype, I assume it is, since normally `if/elif/else` should actually end on `else` as it is logically the progression of that statement chain.

It seems to me like you could either write this as:

```py
if 5 < tier <= 10:
    color = self._MAP_COLORS['mid tier']
if 10 < tier <= 15:
    color = self._MAP_COLORS['high tier']
```

Those get executed sequentially, and there's little need to `elif` them if you're not gonna have a terminating result (which you don't).

The only other readable way to do it is to just write:

```py
if 5 < tier <= 10:
  color = self._MAP_COLORS['mid tier']
# Don't assume one day someone won't do something silly
# and increase map tiers above a certain threshold. Best not to have a random error here
elif 10 < tier:
  self._MAP_COLORS['high tier']
else:
  color = None
```

This way the only time it is None is when the map tier is below 5 (since that is low tier). Contextually even a T16 map needs to be counted as high tier UNLESS there's a very good reason not to do that and I am not aware of it.

Reply:
This had the janky if/elif before I made changes.
I've switched it to if/if, and improved handling around tiers, and excluded vaal temple and shaper guardian maps.